src/AcceptanceTests/Infrastructure/Cleaner.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace NServiceBus.AzureServiceBus.AcceptanceTests.Infrastructure
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Logging;
+    using Microsoft.ServiceBus;
+    using Microsoft.ServiceBus.Messaging;
+    using NServiceBus.AcceptanceTests;
+    using NUnit.Framework;
+
+    public class Cleaner : NServiceBusAcceptanceTest
+    {
+        [Test, Explicit("Intended to be executed explicitly to delete all queues and topics.")]
+        [Category("Cleanup")]
+        public async Task DeleteEntities()
+        {
+            var namespaceManager = NamespaceManager.CreateFromConnectionString(TestUtility.GetDefaultConnectionString());
+
+            var queryQueues = namespaceManager.GetQueuesAsync();
+            var queryTopics = namespaceManager.GetTopicsAsync();
+            await Task.WhenAll(queryQueues, queryTopics)
+                .ConfigureAwait(false);
+
+            var deleteQueues = queryQueues.Result.Select(queueDescription => namespaceManager.DeleteQueueAsync(queueDescription.Path));
+            var deleteTopics = queryTopics.Result.Select(topicDescription => namespaceManager.DeleteTopicAsync(topicDescription.Path));
+
+            await TryWithRetries(Task.WhenAll(deleteQueues.Concat(deleteTopics)), 5)
```

Comment:
Doesn't the namespace manager have a retry policy built in that we could leverage?

Reply:
It's there, but when it comes to exceptions propagated to the client, it won't help us. Throttling or timeouts for example. That's why we do our operations with retries as well.

Reply:
Ok makes sense